internal/pkg/apikey/apikey.go

```diff
@@ -28,6 +33,58 @@ var (
 
 var AuthKey = http.CanonicalHeaderKey("Authorization")
 
+// APIKeyMetadata tracks Metadata associated with an APIKey.
+type APIKeyMetadata struct {
+	ID       string
+	Metadata Metadata
+}
+
+// Read gathers APIKeyMetadata from Elasticsearch using the given client.
+func Read(ctx context.Context, client *elasticsearch.Client, id string) (*APIKeyMetadata, error) {
+	opts := []func(*esapi.SecurityGetAPIKeyRequest){
+		client.Security.GetAPIKey.WithContext(ctx),
+		client.Security.GetAPIKey.WithID(id),
+	}
+
+	res, err := client.Security.GetAPIKey(
+		opts...,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("request to elasticsearch failed: %w", err)
+	}
+	defer res.Body.Close()
+
+	if res.IsError() {
+		return nil, fmt.Errorf("%s: %w", res.String(), ErrAPIKeyNotFound)
+	}
+
+	type APIKeyResponse struct {
+		ID       string   `json:"id"`
+		Metadata Metadata `json:"metadata"`
+	}
+	type GetAPIKeyResponse struct {
+		APIKeys []APIKeyResponse `json:"api_keys"`
+	}
+
+	var resp GetAPIKeyResponse
+	d := json.NewDecoder(res.Body)
+	if err = d.Decode(&resp); err != nil {
```

Comment:
i think this was not fixed so far, when using Decode you need to drain the body e.g `io.Copy(ioutil.Discard, resp.Body)` before closing a body. 
otherwise even though it is Closed it will not release conn for reuse
consider using Unmarshal